network-api/networkapi/buyersguide/management/commands/migrate_products.py

```diff
@@ -0,0 +1,237 @@
+        """For local debugging and testing only."""
+        if settings.DEBUG:
+            message = ' '.join([str(x) for x in messages])
+            print(message)
+
+    def handle(self, *args, **options):
+        products = Product.objects.all()
+        buyersguide_page = self.get_or_create_buyers_guide()
+        for product in products:
+            # 1. Create ProductPage out of this product
+            product = product.specific  # Get the specific class
+
+            # Always refresh the buyersguide_page to update treebeards pathing
+            buyersguide_page.refresh_from_db()
+
+            # Check if ProductPage exists. If it does, continue on.
+            # This check will allow us to run this script more than once if needed
+            if ProductPage.objects.filter(slug=product.slug).exists():
+                self.debug_print("Product already exists... skipping..")
+                continue
+
+            if isinstance(product, SoftwareProduct):
+                self.debug_print("Using a SoftwareProductPage")
+                new_product_page = SoftwareProductPage()
+            elif isinstance(product, GeneralProduct):
+                self.debug_print("Using a GeneralProductPage")
+                new_product_page = GeneralProductPage()
+
+            # Apply the fields that are different or may cause issues if copied directly from one model to another
+            new_product_page.slug_en = product.slug
+            new_product_page.title = product.name
+            new_product_page.title_en = product.name
+            new_product_page.product_url = product.url
+            new_product_page.cloudinary_image = product.cloudinary_image
+            new_product_page.live = not product.draft  # If product is draft, it shall not be live.
+
+            # These are the common fields between SoftwareProductPages and GeneralProductPages
+            common_fields = [
+                'slug', 'privacy_ding', 'adult_content', 'uses_wifi', 'uses_bluetooth',
+                'review_date', 'company', 'blurb', 'price', 'worst_case',
+                'signup_requires_email', 'signup_requires_phone',
+                'signup_requires_third_party_account', 'signup_requirement_explanation',
+                'how_does_it_use_data_collected', 'data_collection_policy_is_bad',
+                'user_friendly_privacy_policy', 'show_ding_for_minimum_security_standards',
+                'meets_minimum_security_standards', 'uses_encryption',
+                'uses_encryption_helptext', 'security_updates', 'security_updates_helptext',
+                'strong_password', 'strong_password_helptext', 'manage_vulnerabilities',
+                'manage_vulnerabilities_helptext', 'privacy_policy', 'privacy_policy_helptext',
+                'phone_number', 'live_chat', 'email', 'twitter'
+            ]
+
+            # Add custom fields based on the product type
+            if isinstance(product, SoftwareProduct):
```

Comment:
Nope, you're correct, I will move the list of specific fields above. We want the Page object created as early as possible so instead of creating the Page object down here, we'll still create it at the top but Ill move the specific-page fields list to the top and remove this secondary `if`